lib/matplotlib/axes/_base.py

```diff
@@ -2013,9 +2013,15 @@ def autoscale_view(self, tight=None, scalex=True, scaley=True):
                 x0, x1 = mtransforms.nonsingular(x0, x1, increasing=False,
                                                  expander=0.05)
             if self._xmargin > 0:
-                delta = (x1 - x0) * self._xmargin
-                x0 -= delta
-                x1 += delta
+                if self.get_xscale() == 'linear':
```

Comment:
In the new implementation the whole `if` statement is gone so it's a moot point.